web/src/features/datasets/server/dataset-router.ts

```diff
@@ -30,6 +30,14 @@ import {
 } from "@/src/features/datasets/server/service";
 import { traceException } from "@langfuse/shared/src/server";
 
+const formatDatasetItemData = (data: string | null | undefined) => {
+  return data === ""
+    ? Prisma.DbNull
+    : !!data
+      ? (JSON.parse(data) as Prisma.InputJsonObject)
```

Comment:
I would recommend to wrap JSON.parse into a try/catch block. Usually any error there produces something that's quite hard to read. 
If you already now it's parseable based on API zod validations the input shouldn't be a string :)